sdk/keyvault/Azure.Security.KeyVault.Keys/Readme.md

```diff
@@ -0,0 +1,282 @@
+        "name": "http://dummy-app-name",
+        "password": "random-password",
+        "tenant": "tenant-ID"
+    }
+    ```
+* Use the above returned credentials information to set **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) environment variables. The following example shows a way to do this in Powershell:
+```cmd
+$Env:AZURE_CLIENT_ID="generated-app-ID"
+$Env:AZURE_CLIENT_SECRET="random-password"
+$Env:AZURE_TENANT_ID="tenant-ID"
+```
+
+* Grant the above mentioned application authorization to perform key operations on the key vault:
+    ```PowerShell
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --key-permissions backup delete get list set
+    ```
+    > --key-permissions:
+    > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+* Use the above mentioned Key Vault name to retrieve details of your Vault which also contains your Key Vault URL:
+    ```PowerShell
+    az keyvault show --name <your-key-vault-name> 
+    ```
+
+#### Create Key client
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables and replaced **your-vault-url** with the above returned URI, you can create the [KeyClient][key_client_class]:
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Keys;
+
+// Create a new key client using the default credential from Azure.Identity
+var client = new KeyClient(vaultUri: <your-vault-url>, credential: new DefaultAzureCredential());
+
+// Create a new key using the key client
+Key key = await Client.CreateKey("key-name", KeyType.EllipticCurve);
+```
+> DefaultAzureCredential():
+> Uses the environment variables previously set (`AZURE_CLIENT_ID`, `AZURE_CLIENT_SECRET`, and `AZURE_TENANT_ID`).
+
+## Key concepts
+### Keys
+Azure Key Vault supports multiple key types and algorithms, and enables the use of Hardware Security Modules (HSM) for high value keys.
+
+### Key Client:
+An asynchronous and synchronous KeyClient client exists in the SDK allowing for selection of a client based on an application's use case. Once you've initialized a KeyClient, you can interact with the primary resource types in Key Vault.
+
+## Examples
+The Azure.Security.KeyVault.Keys package supports synchronous and asynchronous APIs.
+
+The following section provides several code snippets using the [above created](#create-key-client) `client`, covering some of the most common Azure Key Vault Key service related tasks:
+
+### Sync examples
+* [Create a Key](#create-a-key)
+* [Retrieve a Key](#retrieve-a-key)
+* [Update an existing Key](#update-an-existing-key)
+* [Delete a Key](#delete-a-key)
+* [List Keys](#list-keys)
+
+### Async examples
+* [Create a Key](#async-create-a-key)
+
+### Create a Key
+Create a Key to be stored in the Azure Key Vault. If a key with the same name already exists, then a new version of the key is created.
+
+```c#
+// Create a key of any type
```

Comment:
What does 'of any type' mean here?  All the other examples call out the type of the key being created in the comment like `// Create a software Elliptic Curve key`